examples/platform/silabs/SiWx917/SiWx917/sl_wifi_if.cpp

```diff
@@ -632,115 +621,82 @@ static void wfx_rsi_save_ap_info() // translation
-        /* Try to connect Wifi with given Credentials
-         * untill there is a success or maximum number of tries allowed
-         */
-
-        /* Call rsi connect call with given ssid and password
-         * And check there is a success
-         */
-        sl_wifi_credential_t cred;
-        memset(&cred, 0, sizeof(sl_wifi_credential_t));
-        cred.type = SL_WIFI_PSK_CREDENTIAL;
-        memcpy(cred.psk.value, &wfx_rsi.sec.passkey[0], strlen(wfx_rsi.sec.passkey));
-        sl_net_credential_id_t id = SL_NET_DEFAULT_WIFI_CLIENT_CREDENTIAL_ID;
-        status = sl_net_set_credential(id, SL_NET_WIFI_PSK, &wfx_rsi.sec.passkey[0], strlen(wfx_rsi.sec.passkey));
-        if (SL_STATUS_OK != status)
-        {
-            SILABS_LOG("wfx_rsi_do_join: RSI callback register join failed with status: %02x", status);
-            return status;
-        }
-
-        sl_wifi_client_configuration_t ap = { 0 };
-        uint32_t timeout_ms               = 0;
-
-        ap.ssid.length = strlen(wfx_rsi.sec.ssid);
-        memcpy(ap.ssid.value, (int8_t *) &wfx_rsi.sec.ssid[0], ap.ssid.length);
-        ap.security      = connect_security_mode;
-        ap.encryption    = SL_WIFI_NO_ENCRYPTION;
-        ap.credential_id = id;
-        if ((status = sl_wifi_connect(SL_WIFI_CLIENT_INTERFACE, &ap, timeout_ms)) == SL_STATUS_IN_PROGRESS)
-        {
-            callback_status = SL_STATUS_IN_PROGRESS;
-            while (callback_status == SL_STATUS_IN_PROGRESS)
-            {
-                osThreadYield();
-            }
-            status = callback_status;
-        }
-        else
-        {
-            if (is_wifi_disconnection_event || wfx_rsi.join_retries <= WFX_RSI_CONFIG_MAX_JOIN)
-            {
-                SILABS_LOG("wfx_rsi_do_join: Wifi connect failed with status: %x", status);
-                SILABS_LOG("wfx_rsi_do_join: starting JOIN to %s after %d tries\n", (char *) &wfx_rsi.sec.ssid[0],
-                           wfx_rsi.join_retries);
-                wfx_rsi.join_retries += 1;
-                wfx_rsi.dev_state &= ~(WFX_RSI_ST_STA_CONNECTING | WFX_RSI_ST_STA_CONNECTED);
-                wfx_retry_interval_handler(is_wifi_disconnection_event, wfx_rsi.join_retries);
-                if (is_wifi_disconnection_event || wfx_rsi.join_retries <= MAX_JOIN_RETRIES_COUNT)
-                {
-                    event.eventType = WFX_EVT_STA_START_JOIN;
-                    WfxPostEvent(&event);
-                }
-            }
-        }
-    }
+    VerifyOrReturnError(strlen(wfx_rsi.sec.passkey) > SL_WIFI_MAX_PSK_LENGTH, SL_STATUS_SI91X_INVALID_PSK_LENGTH);
+    sl_net_credential_id_t id = SL_NET_DEFAULT_WIFI_CLIENT_CREDENTIAL_ID;
+    status                    = sl_net_set_credential(id, SL_NET_WIFI_PSK, &wfx_rsi.sec.passkey[0], strlen(wfx_rsi.sec.passkey));
+    VerifyOrReturnError(status == SL_STATUS_OK, status);
+
+    uint32_t timeout_ms = 0;
+
+    ap.ssid.length = strlen(wfx_rsi.sec.ssid);
```

Comment:
This is unbounded and unsafe. Please do not use strlen on data that could come from driver/air. Use strnlen.